core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/codesystem/version/CodeSystemVersionCreateRequest.java

```diff
@@ -184,26 +184,25 @@ private void validateEffectiveTime(ServiceProvider context, CodeSystem codeSyste
 			return;
 		}
 
-		Instant mostRecentVersionEffectiveTime = getMostRecentVersionEffectiveDateTime(context, codeSystem);
-		Instant requestEffectiveTime = effectiveTime.atStartOfDay(ZoneOffset.UTC).toInstant();
+		LocalDate mostRecentVersionEffectiveTime = getMostRecentVersionEffectiveDateTime(context, codeSystem);
 
-		if (!requestEffectiveTime.isAfter(mostRecentVersionEffectiveTime)) {
-			throw new BadRequestException("The specified '%s' effective time is invalid. Date should be after '%s'.", requestEffectiveTime, mostRecentVersionEffectiveTime);
+		if (!effectiveTime.isAfter(mostRecentVersionEffectiveTime)) {
+			throw new BadRequestException("The specified '%s' effective time is invalid. Date should be after '%s'.", effectiveTime, mostRecentVersionEffectiveTime);
 		}
 	}
 	
-	private Instant getMostRecentVersionEffectiveDateTime(ServiceProvider context, CodeSystem codeSystem) {
+	private LocalDate getMostRecentVersionEffectiveDateTime(ServiceProvider context, CodeSystem codeSystem) {
 		return CodeSystemRequests.prepareSearchCodeSystemVersion()
 			.all()
 			.filterByCodeSystemShortName(codeSystem.getShortName())
+			.sortBy(SortField.descending(CodeSystemVersionEntry.Fields.EFFECTIVE_DATE))
 			.build(codeSystem.getRepositoryId())
 			.execute(context.service(IEventBus.class))
 			.getSync(1, TimeUnit.MINUTES)
 			.stream()
-			.max(CodeSystemVersionEntry.VERSION_EFFECTIVE_DATE_COMPARATOR)
-			.map(CodeSystemVersionEntry::getEffectiveDate)
-			.map(Instant::ofEpochMilli)
-			.orElse(Instant.EPOCH);
+			.findFirst()
```

Comment:
This search request can now use `.one()` instead of `.all()` and `.findFirst()`.